diff --git a/index.js b/index.js
index ff509874f2bed313f139c537d72f7d01a52b4647..404efdb89089b38bbebe98e744fcab03928149aa 100644
--- a/index.js
+++ b/index.js
@@ -33,7 +33,10 @@ function loader (registryOrVersion) {
     windows['minecraft:merchant'] = { type: protocolId++, inventory: { start: 3, end: 38 }, slots: 39, craft: 2, requireConfirmation: true }
     windows['minecraft:shulker_box'] = { type: protocolId++, inventory: { start: 27, end: 62 }, slots: 63, craft: -1, requireConfirmation: true }
     if (registry.supportFeature('netherUpdateInventoryWindows')) {
-      windows['minecraft:smithing'] = { type: protocolId++, inventory: { start: 3, end: 38 }, slots: 39, craft: 2, requireConfirmation: true }
+      // 1.20 adds the template slot: template 0, base 1, addition 2, result 3
+      windows['minecraft:smithing'] = registry.version['>=']('1.20')
+        ? { type: protocolId++, inventory: { start: 4, end: 39 }, slots: 40, craft: 3, requireConfirmation: true }
+        : { type: protocolId++, inventory: { start: 3, end: 38 }, slots: 39, craft: 2, requireConfirmation: true }
     }
     windows['minecraft:smoker'] = { type: protocolId++, inventory: { start: 3, end: 38 }, slots: 39, craft: 2, requireConfirmation: true }
     windows['minecraft:cartography'] = { type: protocolId++, inventory: { start: 3, end: 38 }, slots: 39, craft: 2, requireConfirmation: true }
